tests/test_metrics.py

```diff
@@ -322,6 +322,31 @@ def test_annual_electricity_generated_by_agent():
     assert_frame_equal(exp, obs)
 
 
+def test_monthly_electricity_generated_by_agent():
+    exp = pd.DataFrame(np.array([
+        (UUID('f22f2281-2464-420a-8325-37320fd418f8'), 1, 0, 100),
+	    (UUID('f22f2281-2464-420a-8325-37320fd418f8'), 1, 1, 100),
+        (UUID('f22f2281-2464-420a-8325-37320fd418f8'), 2, 0, 200),
+	    (UUID('f22f2281-2464-420a-8325-37320fd418f8'), 2, 1, 200),
+	    (UUID('f22f2281-2464-420a-8325-37320fd418f8'), 3, 1, 400)
+	], dtype=ensure_dt_bytes([
+	        ('SimId', 'O'), ('AgentId', '<i8'), ('Year', '<i8'),
+    		('Energy', '<f8')]))
+        )
+    tsp = pd.DataFrame(np.array([
+        (UUID('f22f2281-2464-420a-8325-37320fd418f8'), 1, 3, 1200),
+        (UUID('f22f2281-2464-420a-8325-37320fd418f8'), 2, 3, 2400),
+        (UUID('f22f2281-2464-420a-8325-37320fd418f8'), 1, 12, 1200),
+        (UUID('f22f2281-2464-420a-8325-37320fd418f8'), 2, 12, 2400),
+        (UUID('f22f2281-2464-420a-8325-37320fd418f8'), 3, 12, 4800),
+        ], dtype=ensure_dt_bytes([
+                ('SimId', 'O'), ('AgentId', '<i8'), ('Time', '<i8'),
+                ('Value', '<f8')]))
+        )
+    obs = metrics.annual_electricity_generated_by_agent.func(tsp)
```

Comment:
I think you forgot to update some part.

I think you only have copy pasted the `test_annual_electricity_generated_by_agent()`